### PR TITLE
Fix/rollback 

### PR DESCRIPTION
**Pull request by yosef32**
Allows you to delete a upload object according to its key and not just by a upload ID from S3

### CODE REVIEW

File: proto/file/file.proto [hunk 1]
```diff
@@ -67,6 +68,15 @@ message GetUploadByIDResponse {
   string fileID = 6;
 }
 
+message DeleteUploadByKeyRequest {
```
Comment (Shahar-Y) [inline proto/file/file.proto:71]: It's not just by key, but by key and bucket.
 I'd just call it `DeleteUploadRequest`

File: src/upload/upload.grpc.ts [hunk 2]
```diff
@@ -51,4 +51,11 @@ export class UploadMethods {
     const id: string = call.request.uploadID;
     return UploadService.deleteUpload(id);
   }
+
+  //  Delete an upload from the DB by its file key.
+  public static async DeleteUploadByKey(call: any, callback: any): Promise<void> {
```
Comment (Shahar-Y) [inline src/upload/upload.grpc.ts:56]: Same here. it's not just by key.

File: src/upload/upload.service.ts [hunk 3]
```diff
@@ -122,6 +122,18 @@ export class UploadService {
     }
   }
 
+  /**
+ * Delete an upload from the DB by its file key.
+ * @param key - the key of the file is create when upload created.
+ * @param bucket - the basket key that holds the file.
+ */
+  public static async deleteUploadByKey(key: string, bucket: string): Promise<void> {
+    const deletedUpload = await UploadRepository.deleteByKey(key, bucket);
```
Comment (Shahar-Y) [inline src/upload/upload.service.ts:131]: type

File: proto/file/file.proto (hunk 1 above)
Comment (yosef32) [reply proto/file/file.proto:71]: in update.repository.ts have method updateByKey and she also uses with key and bucket

File: src/upload/upload.grpc.ts (hunk 2 above)
Comment (yosef32) [reply src/upload/upload.grpc.ts:56]: ''

File: src/upload/upload.service.ts (hunk 3 above)
Comment (yosef32) [reply src/upload/upload.service.ts:131]: done

**Review by Shahar-Y — DISMISSED** [review]
Approved, but don't push yet to develop. 
Needs to be pushed with https://github.com/meateam/api-gateway/pull/161

**Comment by yosef32** [comment]
not now

**Review by Shahar-Y — APPROVED** [review]
Looks good :+1:

**Pull request merged**